src/compiler/factory/emitHelpers.ts

```diff
@@ -647,7 +647,7 @@ namespace ts {
                         ar[i] = from[i];
                     }
                 }
-                return to.concat(ar || from);
+                return to.concat(ar || Array.prototype.slice.call(from));

```

Comment:
I don't know how crazy we want to get with this one but we could also do

```suggestion
                return to.concat(ar || (from instanceof Array ? from : Array.prototype.slice.call(from)));
```

Reply:
I considered that, including a check for `isConcatSpreadable`, but it just became long-winded and the performance benefits seemed negligible.

Reply:
I was originally thinking of doing this:
```ts
var __spreadArray = (this && this.__spreadArray) || (function () {
    var toString = Object.prototype.toString;
    var slice = Array.prototype.slice;
    var isArray = typeof Array.isArray === "function" ? Array.isArray : function (obj: any) { return toString.call(obj) === "[object Array]"; };
    var isConcatSpreadable = typeof Symbol === "function" && Symbol.isConcatSpreadable;
    function canConcat(obj) { return isArray(obj) || isConcatSpreadable && obj[isConcatSpreadable]; }
    return function (to, from, pack) {
        if (pack || arguments.length === 2) {
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar) ar = slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
            if (ar) from = ar;
        }
        return to.concat(canConcat(from) ? from : slice.call(from));
    }
})();
```